src/site/content/en/blog/use-long-term-caching/index.md

```diff
@@ -0,0 +1,783 @@
+  practices](https://jakearchibald.com/2016/caching-best-practices/)
+
+## Extract dependencies and runtime into a separate file
+
+### Dependencies
+
+App dependencies tend to change less often than the actual app code. If you move
+them into a separate file, the browser will be able to cache them separately –
+and won’t re-download them each time the app code changes.
+
+Key Term: In webpack terminology, separate files with the app code are called
+_chunks_. We’ll use this name later.
+
+To extract dependencies into a separate chunk, perform three steps:
+
+1. Replace the output filename with `[name].[chunkname].js`:
+
+    ```js
+        // webpack.config.js
+        module.exports = {
+        output: {
+            // Before
+            filename: 'bundle.[chunkhash].js',
+            // After
+            filename: '[name].[chunkhash].js',
+        },
+        };
+    ```
+    
+    When webpack builds the app, it replaces [`[name]`](https://webpack.js.org/configuration/output/#output-filename)
+    with a name of a chunk. If we don’t add the `[name]` part, we’ll have
+    to differentiate between chunks by their hash – which is pretty hard!
+
+1. Convert the `entry` field into an object:
+
+    ```js
+    // webpack.config.js
+    module.exports = {
+    // Before
+    entry: './index.js',
+    // After
+    entry: {
+        main: './index.js',
+    },
+    };
+    ```
+
+    In this snippet, “main” is a name of a chunk. This name will be substituted in
+    place of `[name]` from step 1.
+
+    By now, if you build the app, this chunk will include the whole app code – just
+    like we haven’t done these steps. But this will change in a sec.
+
+1. **In webpack 4**, add the `optimization.splitChunks.chunks: 'all'` option
+    into your webpack config:
+
+    ```js
+    // webpack.config.js (for webpack 4)
+    module.exports = {
+    optimization: {
+        splitChunks: {
+        chunks: 'all',
+        }
+    },
+    };
+    ```
+    
+    This option enables smart code splitting. With it, webpack would extract the vendor code if
+    it gets larger than 30 kB (before minification and gzip). It would also extract the common code –
+    this is useful if your build produces several bundles (e.g.
+    [if you split your app into routes](#split-the-code-into-routes-and-pages)).
+
+    **In webpack 3**, add the [`CommonsChunkPlugin`](https://webpack.js.org/plugins/commons-chunk-plugin/):
+
+    ```js
+    // webpack.config.js (for webpack 3)
+    module.exports = {
+    plugins: [
+        new webpack.optimize.CommonsChunkPlugin({
+        // A name of the chunk that will include the dependencies.
+        // This name is substituted in place of [name] from step 1
+        name: 'vendor',
+
+        // A function that determines which modules to include into this chunk
+        minChunks: module => module.context &&
+            module.context.includes('node_modules'),
+        }),
+    ],
+    };
+    ```
+
```

Comment:
```suggestion
    ```js
    // webpack.config.js (for webpack 3)
    module.exports = {
      plugins: [
        new webpack.optimize.CommonsChunkPlugin({
        // A name of the chunk that will include the dependencies.
        // This name is substituted in place of [name] from step 1
        name: 'vendor',

        // A function that determines which modules to include into this chunk
        minChunks: module => module.context && module.context.includes('node_modules'),
        })
      ]
    };
    ```

```